libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt

```diff
@@ -150,10 +160,10 @@ object MapboxMetricsReporter : MetricsReporter {
     }
 
     private inline fun ifTelemetryIsRunning(func: () -> Unit) {
-        if (enableTelemetry) {
+        if (enableTelemetry && TelemetryUtilsDelegate.getEventsCollectionState()) {
```

Comment:
At first I thought these were the same flags. I think `enableTelemetry` should be renamed into `telemetryInitialized` or sth like that. Not related to this PR tho.